Pspcl/Pspcl.Web/Controllers/StockViewController.cs

```diff
@@ -62,53 +62,88 @@ public JsonResult getMaterialCodes(int materialTypeId)
         }
 
         [HttpPost]
-        public IActionResult AddStock(StockViewModel model, IFormCollection formCollection)
+        public IActionResult AddStock(IFormCollection formCollection)
         {
-            //var formValues = HttpContext.Request.Form;
-            //// Convert the FormCollection to a dictionary
-            //var formData = formValues.ToDictionary(x => x.Key, x => x.Value.ToString());
-            //// Store the serialized dictionary in the temp data
-            //TempData["FormData"] = JsonConvert.SerializeObject(formData);
-
-            model.SelectedMaterialCode = formCollection["selectedMaterialCode"];
-
-            DateTime date = DateTime.Parse(formCollection["GRNDate"]);
-
-            model.GrnDate =date;
-            model.TestReportReference = formCollection["TestReportReference"];
-            model.SelectedMaterialCode = formCollection["materialCode"];
-            model.InvoiceDate = DateTime.Parse(formCollection["InvoiceDate"]);
-            model.InvoiceNumber = formCollection["invoiceNo"];
-            model.SelectedMaterialCode = formCollection["materialCode"];
-            model.EnterRate = decimal.Parse(formCollection["rate"]);
-            model.SelectedMaterialGroup = formCollection["materialGroup"];
-            model.SelectedMaterialType = formCollection["materialType"];
-            model.SelectedRating = formCollection["rating"];
-            model.GrnNumber = long.Parse(formCollection["GrnNO"]);
-            model.PrefixNumber = formCollection["PrefixNumber"];
-
-
-            List<StockMaterial> stockMaterialsList = new List<StockMaterial>();
-
-            for (int i = 12; i < formCollection.Count - 1; i = i + 3)
-            {
-                var element_from = formCollection.ElementAt(i);
-                var element_to = formCollection.ElementAt(i + 1);
-                var element_qty = formCollection.ElementAt(i + 2);
-
-                StockMaterial row = new()
-                {
-                    SerialNumberFrom = Convert.ToInt32(element_from.Value),
-                    SerialNumberTo = int.Parse(element_to.Value),
-                    Quantity = int.Parse(element_qty.Value),
-                };
-                stockMaterialsList.Add(row);
-            }
-
-            model.stockMaterialLList = stockMaterialsList;
-
-            return RedirectToAction("AddStock");
-
+			try
+			{
+				var model = new StockViewModel();
+				//var formValues = HttpContext.Request.Form;
+				//// Convert the FormCollection to a dictionary
+				//var formData = formValues.ToDictionary(x => x.Key, x => x.Value.ToString());
+				//// Store the serialized dictionary in the temp data
+				//TempData["FormData"] = JsonConvert.SerializeObject(formData);
+
+				model.SelectedMaterialCode = formCollection["selectedMaterialCode"];
+
```

Comment:
Why we assigning each value here?

Reply:
actually vikas these things are done by simran and we use form collection that have data in the form of  key value pair that is coming from razor view and we have fetch it to fill our model object . We are mapping data here in the controller.